protocol/nats_jetstream/v3/protocol.go

```diff
@@ -19,6 +19,18 @@ import (
 	"github.com/nats-io/nats.go/jetstream"
 )
 
+type ctxKey string
+
+const (
+	ctxKeySubject ctxKey = "subject"
+)
+
+// WithSubject returns a new context with the subject set to the provided value.
+// This subject will be used when sending or receiving messages and overrides the default.
+func WithAltSubject(ctx context.Context, subject string) context.Context {
```

Comment:
Comment/API is misaligned, suggesting to go with:

```suggestion
func WithSubject(ctx context.Context, subject string) context.Context {
```